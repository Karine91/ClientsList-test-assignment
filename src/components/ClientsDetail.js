import React from "react";
import { Item } from "semantic-ui-react";

const ClientsDetail = props => (
  <Item.Group>
    <Item>
      <Item.Image src={props.general && props.general.avatar} />

      <Item.Content>
        <Item.Header>
          {props.general &&
            `${props.general.firstName} ${props.general.lastName}`}
        </Item.Header>
        <Item.Meta>
          {props.job && `${props.job.title} - ${props.job.company}`}
        </Item.Meta>
        <Item.Description>
          {props.address &&
            `${props.address.zipCode}, ${props.address.country} ${
              props.address.city
            } ${props.address.street}`}
        </Item.Description>
        <Item.Extra>
          {props.contact && `${props.contact.email}, ${props.contact.phone}`}
        </Item.Extra>
      </Item.Content>
    </Item>
  </Item.Group>
);

export default ClientsDetail;
