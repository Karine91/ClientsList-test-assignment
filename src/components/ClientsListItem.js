import React, { Component } from "react";
import { List, Image } from "semantic-ui-react";
import { connect } from "react-redux";
import { setDetail } from "../actions/detail";

export class ClientsListItem extends Component {
  onClickItem = () => {
    const { address, contact, general, job } = this.props;
    this.props.dispatch(setDetail({ address, contact, general, job }));
  };

  render() {
    return (
      <List.Item className="clients-list__item" onClick={this.onClickItem}>
        <Image src={this.props.general.avatar} wrapped size="tiny" />
        <List.Content>
          <List.Header>
            {this.props.general.firstName} {this.props.general.lastName}
          </List.Header>
          <List.Description>{this.props.job.title}</List.Description>
        </List.Content>
      </List.Item>
    );
  }
}

export default connect()(ClientsListItem);
