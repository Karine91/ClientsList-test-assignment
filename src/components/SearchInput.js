import React from "react";
import { connect } from "react-redux";
import { Input } from "semantic-ui-react";
import { setTextFilter } from "../actions/filters";

class SearchInput extends React.Component {
  onChangeSearchInput = e => {
    const value = e.target.value;
    this.props.onChangeSearchInput(value);
  };

  render() {
    return (
      <div className="search-wrapper">
        <Input
          icon="search"
          placeholder="Search..."
          value={this.props.filters.text}
          onChange={this.onChangeSearchInput}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({ filters: state.filters });
const mapDispatchToProps = dispatch => ({
  onChangeSearchInput: text => dispatch(setTextFilter(text))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchInput);
