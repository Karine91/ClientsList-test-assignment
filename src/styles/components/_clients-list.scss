.clients-list {
  &.ui.selection.list > .item {
    border-radius: 0;
    border-bottom: 1px solid #d4d4d5;
  }
}
