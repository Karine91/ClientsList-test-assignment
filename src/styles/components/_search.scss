.search-wrapper {
  border-bottom: 1px solid #d4d4d5;
  .ui.icon.input {
    width: 100%;

    input {
      border: none;
    }
  }
}
