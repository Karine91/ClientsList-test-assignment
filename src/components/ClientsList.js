import React, { Component } from "react";
import { connect } from "react-redux";
import { List } from "semantic-ui-react";
import ClientsListItem from "./ClientsListItem";
import { startSetClients } from "../actions/clients";
import selectClients from "../selectors/clients";

class ClientsList extends Component {
  componentDidMount() {
    this.props.dispatch(startSetClients());
  }
  render() {
    return (
      <List selection className="clients-list">
        {this.props.clients.map((client, index) => (
          <ClientsListItem {...client} key={index} />
        ))}
      </List>
    );
  }
}

const mapStateToProps = state => {
  return {
    clients: selectClients(state.clients, state.filters)
  };
};

export default connect(mapStateToProps)(ClientsList);
