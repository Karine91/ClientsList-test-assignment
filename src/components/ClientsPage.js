import React from "react";
import { connect } from "react-redux";
import { Segment, Container, Grid } from "semantic-ui-react";
import ClientsDetail from "./ClientsDetail";
import ClientsList from "./ClientsList";
import SearchInput from "./SearchInput";

export const ClientsListPage = props => (
  <Container className="main-wrapper">
    <Segment>
      <Grid columns={2} celled="internally">
        <Grid.Column width={6}>
          <SearchInput />
          <ClientsList />
        </Grid.Column>
        <Grid.Column width={10}>
          <ClientsDetail {...props.detail} />
        </Grid.Column>
      </Grid>
    </Segment>
  </Container>
);

const mapStateToProps = ({ detail }) => ({ detail });

export default connect(mapStateToProps)(ClientsListPage);
