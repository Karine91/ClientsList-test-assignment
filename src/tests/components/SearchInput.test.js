import React from "react";
import { shallow } from "enzyme";
import { SearchInput } from "../../components/SearchInput";
import { Input } from "semantic-ui-react";

let wrapper, onChangeSearchInput;

beforeEach(() => {
  onChangeSearchInput = jest.fn();
  wrapper = shallow(
    <SearchInput
      filters={{ text: "" }}
      onChangeSearchInput={onChangeSearchInput}
    />
  );
});

test("should render SearchInput", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should handle text change", () => {
  const event = {
    target: {
      value: "text"
    }
  };
  wrapper.find(Input).prop("onChange")(event);
  expect(onChangeSearchInput).toHaveBeenCalled();
});
