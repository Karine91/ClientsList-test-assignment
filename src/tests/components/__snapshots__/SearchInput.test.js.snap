// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchInput 1`] = `
<div
  className="search-wrapper"
>
  <Input
    icon="search"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</div>
`;
